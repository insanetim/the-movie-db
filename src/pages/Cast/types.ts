import { ErrorMsg } from 'src/interfaces/global.interface'
import { IMovieDetailsEx } from 'src/interfaces/movie.interface'

export type CastHookReturn = {
  error: ErrorMsg
  loading: boolean
  movie: IMovieDetailsEx | undefined
  movieSlug: string
}

export type CastRouteParams = {
  movieSlug: string
}
