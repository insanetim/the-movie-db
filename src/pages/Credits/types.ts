import { ErrorMsg } from 'src/interfaces/global.interface'
import { IPersonDetails } from 'src/interfaces/person.interface'

export enum FilterOptions {
  All = 'all',
  Cast = 'cast',
  Crew = 'crew',
}

export type CreditsHookReturn = {
  dataSource: ICredit[]
  error: ErrorMsg
  handleChangeFilter: (filter: FilterOptions) => void
  isLoading: boolean
  person?: IPersonDetails
  personSlug: string
}

export type CreditsRouteParams = {
  personSlug: string
}

export interface ICredit {
  key: string
  movieSlug: string
  posterPath: null | string
  releaseDate: string
  releaseDateTitle: string
  role: string
  title: string
}
