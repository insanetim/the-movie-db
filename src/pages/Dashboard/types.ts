import { Query } from 'src/interfaces/global.interface'

export type DashboardHook = {
  query: Query
}
