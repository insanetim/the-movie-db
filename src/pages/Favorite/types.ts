import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IMovie, IMoviesList } from 'src/interfaces/movie.interface'

export type FavoriteHookReturn = {
  error: ErrorMsg
  handleMovieDelete: (
    event: React.MouseEvent<HTMLSpanElement>,
    movieId: IMovie['id']
  ) => () => Promise<void>
  handlePagination: HandlePagination
  loading: boolean
  movies: IMoviesList | null
}
