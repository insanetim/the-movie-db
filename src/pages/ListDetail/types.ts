import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IListDetail } from 'src/interfaces/list.interface'
import { IMovie } from 'src/interfaces/movie.interface'

export type ListDetailHook = {
  error: ErrorMsg
  handleListDelete: () => () => Promise<void>
  handleMovieDelete: (
    event: React.MouseEvent<HTMLSpanElement>,
    movieId: IMovie['id']
  ) => () => Promise<void>
  handlePagination: HandlePagination
  list: IListDetail | null
  loading: boolean
}
