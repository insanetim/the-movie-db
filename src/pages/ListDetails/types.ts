import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IListDetails } from 'src/interfaces/list.interface'
import { IMovie } from 'src/interfaces/movie.interface'

export type ListDetailsHookReturn = {
  error: ErrorMsg
  handleListDelete: () => () => Promise<void>
  handleMovieDelete: (
    event: React.MouseEvent<HTMLSpanElement>,
    movieId: IMovie['id']
  ) => () => Promise<void>
  handlePagination: HandlePagination
  isLoading: boolean
  list?: IListDetails
}

export type ListDetailsRouteParams = {
  listSlug: string
}
