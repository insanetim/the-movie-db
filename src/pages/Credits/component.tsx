import { Col, Flex, Image, Row, Segmented, Table, Typography } from 'antd'
import { ColumnsType } from 'antd/es/table'
import { isNil } from 'ramda'
import { Link } from 'react-router-dom'
import NoImage from 'src/assets/images/no-image.svg'
import Empty from 'src/components/UI/Empty'
import Error from 'src/components/UI/Error'
import GoBackLink from 'src/components/UI/GoBackLink'
import Loading from 'src/components/UI/Loading'
import PageTitle from 'src/components/UI/PageTitle'
import {
  releaseDateSorter,
  titleSorter,
} from 'src/utils/helpers/creditsSorters'
import isPresent from 'src/utils/helpers/isPresent'

import useContainer from './hook'
import { FilterOptions, ICredit } from './types'

const Credits: React.FC = () => {
  const { dataSource, error, handleChangeFilter, loading, person, personSlug } =
    useContainer()

  if (loading) {
    return (
      <div className='container top-margin'>
        <Loading />
      </div>
    )
  }

  if (error) {
    return (
      <div className='container top-margin'>
        <Error error={error} />
      </div>
    )
  }

  if (isNil(person)) {
    return <Empty description='Person not found' />
  }

  const columns: ColumnsType<ICredit> = [
    {
      align: 'center',
      dataIndex: 'releaseDateTitle',
      fixed: 'left',
      key: 'releaseDateTitle',
      sorter: releaseDateSorter,
      title: 'Release Date',
      width: 180,
    },
    {
      dataIndex: 'title',
      key: 'title',
      render: (title: string, credit) => {
        return <Link to={`/movie/${credit.movieSlug}`}>{title}</Link>
      },
      sorter: titleSorter,
      title: 'Title',
    },
    {
      dataIndex: 'role',
      key: 'role',
      title: 'Role',
    },
    {
      align: 'center',
      dataIndex: 'posterPath',
      key: 'posterPath',
      render: (posterPath: string, credit) => {
        const POSTER_HEIGHT = 120

        let poster: JSX.Element
        if (isPresent(posterPath)) {
          poster = (
            <Image
              alt={credit.title}
              height={POSTER_HEIGHT}
              src={`https://image.tmdb.org/t/p/w500${posterPath}`}
              style={{ borderRadius: 8 }}
            />
          )
        } else {
          poster = (
            <div
              className='person-poster--no-image'
              style={{ height: POSTER_HEIGHT, margin: '0 auto' }}
            >
              <img
                alt={credit.title}
                src={NoImage}
              />
            </div>
          )
        }

        return poster
      },
      title: 'Poster',
    },
  ]

  return (
    <div className='container top-margin'>
      <GoBackLink
        href={`/person/${personSlug}`}
        title='Back to person details'
      />
      <PageTitle>
        <Typography.Title
          level={1}
          style={{ marginBottom: 0 }}
        >
          {person.name}
        </Typography.Title>
      </PageTitle>
      <Row>
        <Col
          span={24}
          style={{ marginBottom: 12 }}
        >
          <Flex justify='end'>
            <Segmented
              onChange={handleChangeFilter}
              options={Object.values(FilterOptions)}
            />
          </Flex>
        </Col>
        <Col span={24}>
          <Table<ICredit>
            className='credits-table'
            columns={columns}
            dataSource={dataSource}
            pagination={{
              hideOnSinglePage: true,
              pageSize: 5,
              position: ['bottomCenter'],
              showSizeChanger: false,
            }}
            scroll={{ x: 744 }}
          />
        </Col>
      </Row>
    </div>
  )
}

export default Credits
