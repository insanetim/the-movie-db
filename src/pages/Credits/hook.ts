import { addYears } from 'date-fns'
import { groupBy, head, map, pipe, pluck, values } from 'ramda'
import { useMemo, useState } from 'react'
import { useParams } from 'react-router-dom'
import { IPersonCredit } from 'src/interfaces/person.interface'
import { useGetPersonDetailsQuery } from 'src/store/features/person'
import errorMessage from 'src/utils/helpers/errorMessage'
import getIdFromSlug from 'src/utils/helpers/getIdFromSlug'
import getSlug from 'src/utils/helpers/getSlug'

import {
  CreditsHookReturn,
  CreditsRouteParams,
  FilterOptions,
  ICredit,
} from './types'

const useContainer = (): CreditsHookReturn => {
  const { personSlug } = useParams<
    keyof CreditsRouteParams
  >() as CreditsRouteParams
  const personId = getIdFromSlug(personSlug)
  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.All)

  const { data: person, error, isLoading } = useGetPersonDetailsQuery(personId)

  const filteredCredits: IPersonCredit[] = useMemo(() => {
    if (person) {
      const crewCredits = pipe(
        groupBy<IPersonCredit>(item => item.id.toString()),
        values,
        map(credits => {
          const jobs = pluck('job', credits as IPersonCredit[]).join(', ')
          return { ...head(credits as IPersonCredit[]), job: jobs }
        })
      )(person.movie_credits.crew) as IPersonCredit[]

      switch (filter) {
        case 'all':
          return [...person.movie_credits.cast, ...crewCredits]
        case 'cast':
          return [...person.movie_credits.cast]
        case 'crew':
          return [...crewCredits]
      }
    }
    return []
  }, [person, filter])

  const dataSource: ICredit[] = useMemo(() => {
    return filteredCredits.map(credit => ({
      key: `${credit.id}-${credit.character || credit.job}`,
      movieSlug: getSlug(credit.id, credit.title),
      posterPath: credit.poster_path,
      releaseDate:
        credit.release_date || addYears(new Date(), 100).toISOString(),
      releaseDateTitle: credit.release_date
        ? new Date(credit.release_date).getFullYear().toString()
        : '—',
      role: credit.character ? `as ${credit.character}` : credit.job || '—',
      title: credit.title,
    }))
  }, [filteredCredits])

  const handleChangeFilter = (filter: FilterOptions) => {
    setFilter(filter)
  }

  return {
    dataSource,
    error: errorMessage(error),
    handleChangeFilter,
    isLoading,
    person,
    personSlug,
  }
}

export default useContainer
