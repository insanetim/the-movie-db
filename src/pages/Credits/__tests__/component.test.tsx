import { mockPersonDetails, mockedCredits } from 'src/__mocks__/mockPerson'
import renderWithWrapper from 'src/utils/testHelpers/renderWithWrapper'

import Credits from '../component'
import { CreditsHookReturn } from '../types'

const mockedHook: CreditsHookReturn = {
  dataSource: mockedCredits,
  error: null,
  handleChangeFilter: jest.fn(),
  loading: false,
  person: mockPersonDetails,
  personSlug: '1234-test-person',
}
jest.mock('../hook', () => jest.fn(() => mockedHook))

describe('Credits component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<Credits />)

    expect(asFragment()).toMatchSnapshot()
  })

  // it('should match snapshot with other data', () => {
  //   jest.mocked(useContainer).mockReturnValueOnce(
  //     mergeDeepRight(mockedHook, {
  //       movie: {
  //         accountStates: {
  //           favorite: true,
  //           watchlist: true,
  //         },
  //         credits: {
  //           cast: [],
  //           crew: [],
  //         },
  //         release_date: undefined,
  //       },
  //     }) as MovieDetailsHookReturn
  //   )
  //   const { asFragment } = renderWithWrapper(<MovieDetails />)

  //   expect(asFragment()).toMatchSnapshot()
  // })

  it('should match snapshot with empty person', () => {
    mockedHook.person = undefined
    const { asFragment } = renderWithWrapper(<Credits />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with loading', () => {
    mockedHook.loading = true
    const { asFragment } = renderWithWrapper(<Credits />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with error', () => {
    mockedHook.loading = false
    mockedHook.error = 'Something went wrong!'
    const { asFragment } = renderWithWrapper(<Credits />)

    expect(asFragment()).toMatchSnapshot()
  })
})
