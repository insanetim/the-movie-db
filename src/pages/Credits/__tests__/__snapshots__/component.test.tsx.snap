// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credits component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-row css-dev-only-do-not-override-11xg00t"
    >
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-11xg00t"
      >
        <div
          class="ant-typography css-dev-only-do-not-override-11xg00t"
        >
          <a
            href="/person/1234-test-person"
          >
            <span
              aria-label="arrow-left"
              class="anticon anticon-arrow-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="arrow-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
             Back to person details
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-row css-dev-only-do-not-override-11xg00t"
    >
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-11xg00t"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-11xg00t ant-flex-align-center ant-flex-gap-middle"
          style="margin-bottom: 24px;"
        >
          <h1
            class="ant-typography css-dev-only-do-not-override-11xg00t"
            style="margin-bottom: 0px;"
          >
            test-person
          </h1>
        </div>
      </div>
    </div>
    <div
      class="ant-row css-dev-only-do-not-override-11xg00t"
    >
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-11xg00t"
        style="margin-bottom: 12px;"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-11xg00t ant-flex-justify-end"
        >
          <div
            class="ant-segmented css-dev-only-do-not-override-11xg00t"
          >
            <div
              class="ant-segmented-group"
            >
              <label
                class="ant-segmented-item ant-segmented-item-selected"
              >
                <input
                  checked=""
                  class="ant-segmented-item-input"
                  type="radio"
                />
                <div
                  class="ant-segmented-item-label"
                  title="all"
                >
                  all
                </div>
              </label>
              <label
                class="ant-segmented-item"
              >
                <input
                  class="ant-segmented-item-input"
                  type="radio"
                />
                <div
                  class="ant-segmented-item-label"
                  title="cast"
                >
                  cast
                </div>
              </label>
              <label
                class="ant-segmented-item"
              >
                <input
                  class="ant-segmented-item-input"
                  type="radio"
                />
                <div
                  class="ant-segmented-item-label"
                  title="crew"
                >
                  crew
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-11xg00t"
      >
        <div
          class="ant-table-wrapper credits-table css-dev-only-do-not-override-11xg00t"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-11xg00t"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table css-dev-only-do-not-override-11xg00t ant-table-fixed-column ant-table-scroll-horizontal ant-table-has-fix-left"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                    style="overflow-x: auto; overflow-y: hidden;"
                  >
                    <table
                      style="width: 744px; min-width: 100%; table-layout: fixed;"
                    >
                      <colgroup>
                        <col
                          style="width: 180px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            aria-label="Release Date"
                            class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last ant-table-column-has-sorters"
                            scope="col"
                            style="position: sticky; left: 0px; text-align: center;"
                            tabindex="0"
                          >
                            <div
                              class="ant-table-column-sorters"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Release Date
                              </span>
                              <span
                                class="ant-table-column-sorter ant-table-column-sorter-full"
                              >
                                <span
                                  aria-hidden="true"
                                  class="ant-table-column-sorter-inner"
                                >
                                  <span
                                    aria-label="caret-up"
                                    class="anticon anticon-caret-up ant-table-column-sorter-up"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-up"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                  <span
                                    aria-label="caret-down"
                                    class="anticon anticon-caret-down ant-table-column-sorter-down"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            aria-label="Title"
                            class="ant-table-cell ant-table-column-has-sorters"
                            scope="col"
                            tabindex="0"
                          >
                            <div
                              class="ant-table-column-sorters"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Title
                              </span>
                              <span
                                class="ant-table-column-sorter ant-table-column-sorter-full"
                              >
                                <span
                                  aria-hidden="true"
                                  class="ant-table-column-sorter-inner"
                                >
                                  <span
                                    aria-label="caret-up"
                                    class="anticon anticon-caret-up ant-table-column-sorter-up"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-up"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                  <span
                                    aria-label="caret-down"
                                    class="anticon anticon-caret-down ant-table-column-sorter-down"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Role
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                            style="text-align: center;"
                          >
                            Poster
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          aria-hidden="true"
                          class="ant-table-measure-row"
                          style="height: 0px; font-size: 0px;"
                        >
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1234-the-chosen-one"
                        >
                          <td
                            class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last"
                            style="position: sticky; left: 0px; text-align: center;"
                          >
                            1999
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <a
                              href="/movie/1234-test-movie"
                            >
                              the-matrix
                            </a>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            the-chosen-one
                          </td>
                          <td
                            class="ant-table-cell"
                            style="text-align: center;"
                          >
                            <div
                              class="ant-image css-dev-only-do-not-override-11xg00t"
                              style="height: 120px;"
                            >
                              <img
                                alt="the-matrix"
                                class="ant-image-img css-dev-only-do-not-override-11xg00t"
                                height="120"
                                src="https://image.tmdb.org/t/p/w500/image"
                                style="height: 120px; border-radius: 8px;"
                              />
                              <div
                                class="ant-image-mask"
                              >
                                <div
                                  class="ant-image-mask-info"
                                >
                                  <span
                                    aria-label="eye"
                                    class="anticon anticon-eye"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="eye"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                                      />
                                    </svg>
                                  </span>
                                  Preview
                                </div>
                              </div>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="4321-darth-maul"
                        >
                          <td
                            class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last"
                            style="position: sticky; left: 0px; text-align: center;"
                          >
                            2020
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <a
                              href="/movie/4321-test-movie"
                            >
                              star-wars
                            </a>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            darth-maul
                          </td>
                          <td
                            class="ant-table-cell"
                            style="text-align: center;"
                          >
                            <div
                              class="person-poster--no-image"
                              style="height: 120px; margin: 0px auto;"
                            >
                              <img
                                alt="star-wars"
                                src="test-file-stub"
                              />
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Credits component should match snapshot with empty person 1`] = `
<DocumentFragment>
  <div
    class="css-dev-only-do-not-override-11xg00t ant-empty ant-empty-normal"
  >
    <div
      class="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#f5f5f5"
            rx="32"
            ry="7"
          />
          <g
            fill-rule="nonzero"
            stroke="#d9d9d9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#fafafa"
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      class="ant-empty-description"
    >
      Person not found
    </div>
  </div>
</DocumentFragment>
`;

exports[`Credits component should match snapshot with error 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-result ant-result-error css-dev-only-do-not-override-11xg00t"
    >
      <div
        class="ant-result-icon"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            fill-rule="evenodd"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
            />
          </svg>
        </span>
      </div>
      <div
        class="ant-result-title"
      >
        Something went wrong!
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Credits component should match snapshot with loading 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-row ant-row-center css-dev-only-do-not-override-11xg00t"
    >
      <div
        class="ant-col css-dev-only-do-not-override-11xg00t"
      >
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-spinning css-dev-only-do-not-override-11xg00t"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
